presto-hive/src/main/java/com/facebook/presto/hive/SyncPartitionMetadataProcedure.java

```diff
@@ -127,8 +141,17 @@ private void doSyncPartitionMetadata(ConnectorSession session, String schemaName
 
         try {
             FileSystem fileSystem = hdfsEnvironment.getFileSystem(context, tableLocation);
-            List<String> partitionsInMetastore = metastore.getPartitionNames(new MetastoreContext(session.getIdentity(), session.getQueryId(), session.getClientInfo(), session.getSource(), getMetastoreHeaders(session), isUserDefinedTypeEncodingEnabled(session), metastore.getColumnConverterProvider()), schemaName, tableName)
+            List<String> partitionNamesInMetastore = metastore.getPartitionNames(metastoreContext, schemaName, tableName)
                     .orElseThrow(() -> new TableNotFoundException(schemaTableName));
+            List<String> partitionsInMetastore = new ArrayList<>(partitionNamesInMetastore.size());
```

Comment:
Use ImmutableList and ImmutableList.builder().builder()

Reply:
fixed